stubs/synapse/synapse_rust/push.pyi

```diff
@@ -0,0 +1,30 @@
+from typing import Any, Collection, Dict, Mapping, Sequence, Tuple, Union
+
+from synapse.types import JsonDict
+
+class PushRule:
+    rule_id: str
+    priority_class: int
+    conditions: Sequence[Mapping[str, str]]
+    actions: Sequence[Union[Mapping[str, Any], str]]
+    default: bool
+    default_enabled: bool
+
+    @staticmethod
+    def from_db(
+        rule_id: str, priority_class: int, conditions: str, actions: str
+    ) -> "PushRule": ...
+
+class PushRules:
+    def __init__(self, rules: Collection[PushRule]): ...
+    def rules(self) -> Collection[PushRule]: ...
+
+class FilteredPushRules:
+    def __init__(
+        self,
+        push_rules: PushRules,
+        enabled_map: Dict[str, bool],
+        msc3786_enabled: bool,
+        msc3772_enabled: bool,
+    ): ...
+    def rules(self) -> Collection[Tuple[PushRule, bool]]: ...
```

Comment:
Naively: do these need to be classes? Naively they look like they could be (exposed to Python as?) free functions, given that they have just __init__ and one method.

Reply:
Good Q. We want to be able to pass around the `FilteredPushRules` and pass them into rust functions (in future PR), so I think we do want them to be actual classes.

Reply:
they have extra functions in Rustland so it might make sense to keep them as classes, even though you can't see everything about them in Python